File: Solutions/0014_LongestCommonPrefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string prefix = "";

        int position = 0;

        int n = strs.size();

        bool isChecking = true;
        
        for(int p = 0; p < strs[0].length(); p++) {
            char curr = strs[0][p];
            for(int i = 1; i < n; i++) {
                if(strs[i][p] != curr)
                    return prefix;
            }

            prefix += curr;
        }

        return prefix;
    }
};

File: Solutions/0014_LongestCommonPrefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        _prefix = ""
        if len(strs) == 0:
            return _prefix
        if len(strs[0]) == 0:
            return _prefix

        _prefix_candidate = strs[0][0]
        while True:
            for _s in strs:
                if len(_s) == 0:
                    return _prefix
                if _prefix_candidate != _s[0]:
                    return _prefix
            
            _prefix += _prefix_candidate
            for i in range(len(strs)):
                strs[i] = strs[i][1:]
                if len(strs[i]) == 0:
                    return _prefix
                
            _prefix_candidate = strs[0][0]